pkg/nodeagent/controller/operatingsystemconfig/add.go

```diff
@@ -81,3 +96,52 @@ func (r *Reconciler) SecretPredicate() predicate.Predicate {
 		GenericFunc: func(_ event.GenericEvent) bool { return false },
 	}
 }
+
+func reconcileRequest(obj client.Object) reconcile.Request {
+	return reconcile.Request{NamespacedName: types.NamespacedName{
+		Name:      obj.GetName(),
+		Namespace: obj.GetNamespace(),
+	}}
+}
+
+// RandomDurationWithMetaDuration is an alias for `utils.RandomDurationWithMetaDuration`. Exposed for unit tests.
+var RandomDurationWithMetaDuration = utils.RandomDurationWithMetaDuration
+
+// EnqueueWithJitterDelay returns handler.Funcs which enqueues the object with a random jitter duration for 'update'
+// events. 'Create' events are enqueued immediately.
+func (r *Reconciler) EnqueueWithJitterDelay(log logr.Logger) handler.EventHandler {
+	return &handler.Funcs{
+		CreateFunc: func(_ context.Context, evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+			if evt.Object == nil {
+				return
+			}
+			q.Add(reconcileRequest(evt.Object))
+		},
+
+		UpdateFunc: func(_ context.Context, evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+			oldSecret, ok := evt.ObjectOld.(*corev1.Secret)
+			if !ok {
+				return
+			}
+			newSecret, ok := evt.ObjectNew.(*corev1.Secret)
+			if !ok {
+				return
+			}
+
+			_, _, oldOSCChecksum, err := extractOSCFromSecret(oldSecret)
+			if err != nil {
+				log.Error(err, "Failed to get OSC checksum for old secret")
+				return
+			}
+			_, _, newOSCChecksum, err := extractOSCFromSecret(newSecret)
+			if err != nil {
+				log.Error(err, "Failed to get OSC checksum for new secret")
+				return
+			}
+
+			if oldOSCChecksum != newOSCChecksum {
+				q.AddAfter(reconcileRequest(evt.ObjectNew), RandomDurationWithMetaDuration(r.Config.SyncJitterPeriod))
+			}
```

Comment:
Why do we need to compare the checksums? Isn't it sufficient to check for changes in `osc.yaml` key as `SecretPredicate` is doing?
A downside of this approach is that if the the OSC in the secret becomes invalid somehow, an update with a corrected OSC would not start a reconciliation because the checksum calculation of the old secret fails.